File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()

        while n != 1 and n not in seen:
            seen.add(n)
            n = self.next(n)

        return n == 1

    def next(self, n: int) -> int:
        sum_val = 0

        while n > 0:
            digit = n % 10
            sum_val += digit * digit
            n //= 10 # Force integer floor division

        return sum_val

File: 0202-happy-number/0202-happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
    const seen = new Set();

    while (n !== 1 && !seen.has(n)) {
        seen.add(n);
        n = getNextSquareSum(n);
    }

    return n === 1;
};

// Helper function to extract digits and sum their squares
function getNextSquareSum(n) {
    let sum = 0;

    while (n > 0) {
        const digit = n % 10;
        sum += digit * digit;
        n = Math.floor(n / 10); // Safe integer floor division
    }

    return sum;
}
